### ISSUE TITLE
Mod suggestion when dungeons arise 

### ISSUE THREAD

**Issue by 0ldsandwich5537** [EnigmaticaModpacks/Enigmatica6#3992]
### URL

https://www.curseforge.com/minecraft/mc-mods/when-dungeons-arise

### Why would you like the mod added?

A fun mod to add more into exploration along with more difficult dungeons to raid. Should be rarer so it's something more special to find in the world.

**Comment by NillerMedDild** [comment]
Thanks for the suggestion. I tried it once with Terralith and the worldgen lag was unbearable, I'll try again without :)

**Comment by NillerMedDild** [comment]
Oh I thought this suggestion was for E8, my bad. E6 has plenty of dungeons already :)

**Comment by 0ldsandwich5537** [comment]
Fair enough just wanted to throw it out there didn't feel like is enough "surface coverage" most dungeons are underground

**Comment by NillerMedDild** [comment]
Pretty good point, but I'm afraid it'd tank performance :(

**Comment by 0ldsandwich5537** [comment]
I've got it running on my server I haven't noticed any issues so far just thought I would suggest to have pack support instead of having to update and that separately.

**Comment by NillerMedDild** [comment]
I'll have another look when I get some extra time 👍 reports from players, up/downvotes to the main post is very welcome here.

**Comment by juh9870** [comment]
Installed this mod to try it out and it works fine. The only issue I faced was lag spike upon opening lootr chest with structure map in it (possibly because of structure location). I'd like to see this mod in e6 too, but I think some loot tables require tweaking. Aka, loot of "final" chests of some dungeons is similar in nature but much worse than "final" loot chests of Dungeon Crawl dungeon, despite Dungeons Arise dungeons being much harder

**Comment by juh9870** [comment]
After a bit more of testing, I'd say we should disable underground dungeons, while only keeping surface/aerial/water ones. E6 already have enough underground structures as is but not enough surface coverage, so it seems reasonable to populate one without affecting other

**Comment by 0ldsandwich5537** [comment]
> 

agree with this 100%
due to the lower water level some of the underground dungeons are visible in rivers and in some areas on the surface.
along with that quark caves eat into the dungeons.
also some of the other dungeons overlap just due to the overall size of the dungeons arise dungeons.

on another note after talking to the mod author he says he plans to make the dungeons themselves have internal progression. as in better loot as you progress further into the dungeons along with more difficult mobs. no precise idea as to when that will happen though.
![2022-01-13_14 37 28](https://user-images.githubusercontent.com/61948501/149859876-9914bc4e-d164-4907-b811-ea23787257db.png)
![2022-01-13_13 09 48](https://user-images.githubusercontent.com/61948501/149859931-eb40a0f3-208a-4817-9a3c-f1b33cb327ca.png)

**Comment by novaexe** [comment]
causes worldgen deadlock/indefinite single player server hang

tested on current dev build for E6E with no alterations to the pack

tried switching to release versions of the mod instead of beta with 0 effect and removing the mod completely solved the issue

see support in discord (today at 12:29pm) for the entire conversation on this issue

**Comment by novaexe** [comment]
sorry unfortunately it never truly crashed so the log doesn't say much other than spark profile, i reinstalled the mod from curseforge directly and haven't had the issue since so i'm assuming corrupted mod from the gitupdater or something like that. However it did cause 2 infinite crash loops and i had to revert from backup and it's a miracle i noticed the massive dungeon in the chunks that did load and assumed it was dungeons arise causing the issue in some way. between the talk's of myself Thebooboo and TelepathicGrunt i have found that even though the issue is resolved dungeons arise causes most if not all of my massive tps spikes (upwards of 1000ms). so I would watch this mod carefully and as boo suggested perhaps make the dungeons a tad smaller because it's clearly evident (in my instance) that this mod absolutely devours performance in the best case and can cause indefinite hangs/deadlocks in the worst case. If it happens again though telepathicgrunt suggested i get VisualVM so next time ill have a whole thread dump + blame mod installed to post :)

**Comment by NillerMedDild** [comment]
Alright, perhaps the best cause of action would be for us to disable the largest structures, or try to persuade them to use a more efficient jigsaw system.

**Comment by theboo** [comment]
we do have control over the number of jigsaw iterations in any given structure from that mod, could tone down some of the massive ones as well

**Comment by juh9870** [comment]
Underground structures often collide between themselves, or conflict with underground generation from other mods (especially Quark caves), so I'd recommend disabling underground structures at all. E6 have enough underground coverage already, imo.
